src/daemon/keymap.c

```diff
@@ -829,6 +861,87 @@ void process_input_urb(void* context, unsigned char* buffer, int urblen, ushort
                                 if(matched)
                                     CLEAR_KEYBIT(targetkb->input.keys, 122);
                             }
+                        }else if(IS_K70CORERGB(targetkb)){
+                            #define K70_CORE_MEDIA_MASK 0x60
+                            if(buffer[1] == 0x5 && buffer[2] == K70_CORE_MEDIA_MASK) {
+                                // This is a packet from the volume knob
+                                // We need to handle it appropriately then clear the keys so we stop getting garbage
+                                if(buffer[4] == 0x01) {
+                                    // Volume knob turned right
+                                    // Store
+                                    uchar media_key_state_1 = targetkb->input.keys[12];
+                                    uchar media_key_state_2 = targetkb->input.keys[16];
+
+                                    // Apply fresh key data
+                                    memset(targetkb->input.keys, 0, sizeof targetkb->input.keys);
```

Comment:
Zeroing out the keys will let go of any keys that were held down.

Reply:
This is the thing that fixed the media dial sending garbled text and breaking the keyboard. I'll see if I can avoid clearing it but I think this is the thing that made it work. Note: This is only for the media dial's volume up/down which shouldn't be used with held keys.

Reply:
I think this is similar to the media keys getting stuck down bug you fixed on the Strafe. I used similar code that maybe could be exactly the same.